client/state/sharing/keyring/reducer.js

```diff
@@ -28,12 +28,18 @@ export const isFetching = createReducer( false, {
 // Stores the list of available keyring connections
 export const items = createReducer( {}, {
 	[ KEYRING_CONNECTIONS_RECEIVE ]: ( state, { connections } ) => ( { ...keyBy( connections, 'ID' ) } ),
-	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: [ ...state[ ID ].sites, siteId.toString() ]
-	} } ),
-	[ PUBLICIZE_CONNECTION_DELETE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: without( state[ ID ].sites, siteId.toString() )
-	} } ),
+	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { connection: { keyring_connection_ID: id, site_ID } } ) => {
+		const connection = state[ id ];
+		const sites = [ ...connection.sites, site_ID.toString() ];
```

Comment:
The `/me/keyring-connections` endpoint returns the site IDs as an array of strings. While `user_ID` gets treated with an `absint()` before being returned, site IDs do not.